channel/channel.ml

```diff
@@ -25,6 +25,10 @@ module Make(Flow:V1_LWT.FLOW) = struct
   type +'a io = 'a Lwt.t
   type 'a io_stream = 'a Lwt_stream.t
 
+  exception End_of_file (* at least one user understands this exception *)
```

Comment:
That code is doing two horrible things: using an exception to represent the (non-exceptional, should-be-handled) case of end of file, and then converting it to the magic value `""` to mean the same thing.

The cohttp docs say:

> It returns an empty string if EOF *or some other error condition* occurs on the input channel

Would be good to standardise on something a bit saner everywhere...